// @aws-lambda-powertools/tracer/lib/config/EnvironmentVariablesService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentVariablesService = void 0;
const commons_1 = require("@aws-lambda-powertools/commons");
class EnvironmentVariablesService extends commons_1.EnvironmentVariablesService {
    constructor() {
        super(...arguments);
        // Environment variables
        this.awsExecutionEnv = 'AWS_EXECUTION_ENV';
        this.samLocalVariable = 'AWS_SAM_LOCAL';
        this.tracerCaptureErrorVariable = 'POWERTOOLS_TRACER_CAPTURE_ERROR';
        this.tracerCaptureHTTPsRequestsVariable = 'POWERTOOLS_TRACER_CAPTURE_HTTPS_REQUESTS';
        this.tracerCaptureResponseVariable = 'POWERTOOLS_TRACER_CAPTURE_RESPONSE';
        this.tracingEnabledVariable = 'POWERTOOLS_TRACE_ENABLED';
    }
    getAwsExecutionEnv() {
        return this.get(this.awsExecutionEnv);
    }
    getCaptureHTTPsRequests() {
        return this.get(this.tracerCaptureHTTPsRequestsVariable);
    }
    getSamLocal() {
        return this.get(this.samLocalVariable);
    }
    getTracingCaptureError() {
        return this.get(this.tracerCaptureErrorVariable);
    }
    getTracingCaptureResponse() {
        return this.get(this.tracerCaptureResponseVariable);
    }
    getTracingEnabled() {
        return this.get(this.tracingEnabledVariable);
    }
}
exports.EnvironmentVariablesService = EnvironmentVariablesService;
